File: CH-0(Python_vs_C++)/oops.cpp
/* 
Yes, in addition to accessing member variables, the dot operator can also be used to access member functions and to add or modify object attributes in both C++ and Python.

In C++, you can use the dot operator to call member functions of an object, and to access or modify its member variables, like this:
*/
#include <iostream>
#include <string>
using namespace std;

class Person {
public:
    string name;
    int age;

    void say_hello() {
        cout << "Hello, my name is " << name << " and I'm " << age << " years old." << endl;
    }
};

int main() {
    Person person;
    person.name = "John";
    person.age = 30;

    // Call the "say_hello" method using the dot operator
    person.say_hello();  // Output: "Hello, my name is John and I'm 30 years old."

    // Access or modify the "name" member variable using the dot operator
    cout << person.name << endl;   // Output: "John"
    person.name = "Mary";
    cout << person.name << endl;   // Output: "Mary"

    return 0;
}

File: CH-0(Python_vs_C++)/oops.py
""" 
In Python, you can use the dot operator to call methods of an object, and to add or modify its attributes, like this:
"""

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def say_hello(self):
        print(f"Hello, my name is {self.name} and I'm {self.age} years old.")

person = Person("John", 30)

# Call the "say_hello" method using the dot operator
person.say_hello()  # Output: "Hello, my name is John and I'm 30 years old."

# Access or modify the "name" attribute using the dot operator
print(person.name)   # Output: "John"
person.name = "Mary"
print(person.name)   # Output: "Mary"

# Add a new attribute using the dot operator
person.height = 175
print(person.height)  # Output: 175

